File: src/lib/auth-form.test.ts
import assert from "node:assert/strict";
import { describe, it } from "node:test";
import { confirmFieldError, emailFieldError, passwordFieldError } from "./auth-form.ts";

describe("auth form validation", () => {
  it("requires a real-looking email", () => {
    assert.equal(emailFieldError(""), "Enter your email.");
    assert.equal(emailFieldError("not-an-email"), "Enter a valid email address.");
    assert.equal(emailFieldError("host@example.com"), null);
  });

  it("surfaces the 8+ character rule only when creating", () => {
    assert.equal(passwordFieldError("", { creating: true }), "Choose a password.");
    assert.equal(passwordFieldError("short", { creating: true }), "Use at least 8 characters.");
    assert.equal(passwordFieldError("longenough", { creating: true }), null);
    assert.equal(passwordFieldError("", { creating: false }), "Enter your password.");
    assert.equal(passwordFieldError("short", { creating: false }), null);
  });

  it("checks password confirmation", () => {
    assert.equal(confirmFieldError("password1", ""), "Confirm your password.");
    assert.equal(confirmFieldError("password1", "password2"), "Those passwords do not match.");
    assert.equal(confirmFieldError("password1", "password1"), null);
  });
});

File: android/core/src/test/kotlin/com/eventview/core/AuthFormTest.kt
package com.eventview.core

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNull

class AuthFormTest {
    @Test
    fun requiresRealLookingEmail() {
        assertEquals("Enter your email.", AuthForm.emailFieldError(""))
        assertEquals("Enter a valid email address.", AuthForm.emailFieldError("not-an-email"))
        assertNull(AuthForm.emailFieldError("host@example.com"))
    }

    @Test
    fun surfacesEightCharacterRuleOnlyWhenCreating() {
        assertEquals("Choose a password.", AuthForm.passwordFieldError("", creating = true))
        assertEquals("Use at least 8 characters.", AuthForm.passwordFieldError("short", creating = true))
        assertNull(AuthForm.passwordFieldError("longenough", creating = true))
        assertEquals("Enter your password.", AuthForm.passwordFieldError("", creating = false))
        assertNull(AuthForm.passwordFieldError("short", creating = false))
    }

    @Test
    fun checksPasswordConfirmation() {
        assertEquals("Confirm your password.", AuthForm.confirmFieldError("password1", ""))
        assertEquals("Those passwords do not match.", AuthForm.confirmFieldError("password1", "password2"))
        assertNull(AuthForm.confirmFieldError("password1", "password1"))
    }

    @Test
    fun displayNameFallsBackToEmailLocalPart() {
        assertEquals("Ada", AuthForm.displayName("Ada", "ada@example.com"))
        assertEquals("ada", AuthForm.displayName("  ", "ada@example.com"))
        assertEquals("Host", AuthForm.displayName("", ""))
    }
}
